src/network/postgres_protocol_handler.cpp

```diff
@@ -316,21 +316,21 @@ ProcessResult PostgresProtocolHandler::ExecQueryMessage(InputPacket *pkt, const
         statement_name = tokens.at(1);
         auto statement_cache_itr = statement_cache_.find(statement_name);
         if (statement_cache_itr != statement_cache_.end()) {
-          statement_ = *statement_cache_itr;
+          traffic_cop_->statement_ = *statement_cache_itr;
         }
         // Did not find statement with same name
         else {
-          error_message_ = "The prepared statement does not exist";
-          LOG_ERROR("%s", error_message_.c_str());
+          traffic_cop_->error_message_ = "The prepared statement does not exist";
+          LOG_ERROR("%s", traffic_cop_->error_message_.c_str());
           SendErrorResponse(
-              {{NetworkMessageType::HUMAN_READABLE_ERROR, error_message_}});
+              {{NetworkMessageType::HUMAN_READABLE_ERROR, traffic_cop_->error_message_}});
           SendReadyForQuery(NetworkTransactionStateType::IDLE);
           return ProcessResult::COMPLETE;
         }
 
-        query_type_ = statement_->GetQueryType();
-        query_ = statement_->GetQueryString();
-        std::vector<int> result_format(statement_->GetTupleDescriptor().size(), 0);
+        traffic_cop_->query_type_ = traffic_cop_->statement_->GetQueryType();
```

Comment:
If the query type and query can be accessed through the statement object, why can not we directly just use the statement object? That way we can use less field and have a better chance get rid of the inconsistency.